x-pack/plugins/fleet/server/services/epm/elasticsearch/template/default_settings.ts

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { appContextService } from '../../../app_context';
+import type { Field, Fields } from '../../fields/field';
+
+const QUERY_DEFAULT_FIELD_TYPES = ['keyword', 'text'];
+const QUERY_DEFAULT_FIELD_LIMIT = 1024;
+
+const flattenFieldsToNameAndType = (
+  fields: Fields,
+  path: string = ''
+): Array<Pick<Field, 'name' | 'type'>> => {
+  let newFields: Array<Pick<Field, 'name' | 'type'>> = [];
+  fields.forEach((field) => {
+    const fieldName = path ? `${path}.${field.name}` : field.name;
+    newFields.push({
+      name: fieldName,
+      type: field.type,
+    });
+    if (field.fields && field.fields.length) {
+      newFields = newFields.concat(flattenFieldsToNameAndType(field.fields, fieldName));
+    }
+  });
+  return newFields;
+};
+
+export function buildDefaultSettings({
+  templateName,
+  packageName,
+  fields,
+  ilmPolicy,
+  type,
+}: {
+  type: string;
```

Comment:
Can `type` here only be `logs` or `metrics`? Should we express that in the type?

Reply:
`datastream.type` is not  typed to only logs and metrics, I think there is some other options too `synthetics` for example, I would keep this as string for now.